http-proxy-configurator-operator/tests/unit/test_state.py

```diff
@@ -0,0 +1,272 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for the state module."""
+
+from unittest.mock import Mock
+
+import pytest
+from ops import CharmBase
+
+import state
+
+
+def test_state_from_charm():
+    """
+    arrange: mock a charm with valid proxy configuration
+    act: instantiate a State
+    assert: the data matches the charm configuration
+    """
+    charm = Mock(CharmBase)
+    charm.config = {
+        "http-proxy-domains": "example.com,test.org",
+        "http-proxy-auth": "none,srcip",
+        "http-proxy-source-ips": "192.168.1.1,10.0.0.1",
+    }
+    charm_state = state.State.from_charm(charm)
+
+    assert charm_state.http_proxy_domains == charm.config.get("http-proxy-domains").split(",")
+    assert charm_state.http_proxy_auth == [
+        state.ProxyAuthMethod.NONE,
+        state.ProxyAuthMethod.SRC_IP,
+    ]
+    assert [str(ip) for ip in charm_state.http_proxy_source_ips] == charm.config.get(
+        "http-proxy-source-ips"
+    ).split(",")
+
+
+def test_state_from_charm_all_auth_methods():
+    """
+    arrange: mock a charm with all valid auth methods
+    act: instantiate a State
+    assert: all auth methods are correctly parsed
+    """
+    charm = Mock(CharmBase)
+    charm.config = {
+        "http-proxy-domains": "example.com",
+        "http-proxy-auth": "none,srcip,userpass,srcip+userpass",
+        "http-proxy-source-ips": "192.168.1.1",
+    }
+    charm_state = state.State.from_charm(charm)
+
+    assert charm_state.http_proxy_auth == [
+        state.ProxyAuthMethod.NONE,
+        state.ProxyAuthMethod.SRC_IP,
+        state.ProxyAuthMethod.USERPASS,
+        state.ProxyAuthMethod.SRC_IP_AND_USERPASS,
+    ]
+
+
+def test_state_from_charm_empty_source_ips():
+    """
+    arrange: mock a charm with empty source IPs
+    act: instantiate a State
+    assert: source IPs list is empty
+    """
+    charm = Mock(CharmBase)
+    charm.config = {
+        "http-proxy-domains": "example.com",
+        "http-proxy-auth": "none",
+        "http-proxy-source-ips": "",
+    }
+    charm_state = state.State.from_charm(charm)
+
+    assert not charm_state.http_proxy_source_ips
+
+
+def test_state_from_charm_no_domains():
```

Comment:
You can probably merge all these tests that expect an exception with pytest.parametrize